### ISSUE TITLE
Focus indicator missing on link style button

### ISSUE THREAD

**Issue by shainanigans** [StackExchange/Stacks#996]
**Describe the bug**
There is no focus state indicator for buttons with the class `.s-btn__link`. Per WCAG guidelines, all focusable elements must have a visual focus indicator when in focus.

**To Reproduce**
1. Go to [documentation > Buttons > Reset](https://stackoverflow.design/product/components/buttons/#resets)
2. Using keyboard navigation only, tab to the "Link button" example.
3. See that there is no indication that the button is in focus

**Expected behavior**
The focus state of a button with `.s-btn__link` is the same as a default link (using an anchor tag).

**Screenshots**
The link's focus state:
<img width="193" alt="Screen Shot 2022-06-27 at 5 11 46 PM" src="https://user-images.githubusercontent.com/14816224/176036468-3339c9a3-6462-4af2-9882-8e8b9417f0af.png">

**Additional context**
In addition to in-line text use cases for a button that has been styled this way,`.s-btn__link` is also used in the header in Teams under the help menu for "Getting Started." All other items in that menu are anchor tags so behave as expected.
